Initial doc changes for settings/serverless-scan

KICKdesigns opened the pull request.

iansk reviewed (changes requested):
@KICKdesigns I provided a lot of feedback in the PR for `/settings/registry` (https://github.com/twistlock/docs/pull/367). This endpoint works in the same way in general. Please update accordingly.

iansk commented:
@KICKdesigns The way these endpoints work will change in Hamilton. The purpose of the change is to make it easier for customers to scope the serverless functions to be scanned

Can you please re-examine, re-work, and re-test these endpoints against the build I sent you? You'll also need to update this PR.

I'll follow with more details via email.

Thanks!

KICKdesigns commented:
@iansk The PR has been updated for the Hamilton build. The params spreadsheet linked in the tracking doc will have a tab for the Hamilton build. The main changes are:

- Removed some fields from shared.ServerlessScanSpecification which are no longer in Hamilton's swagger
- Added (missing) common.CloudProvider fields
- Updated description of "awsRegionType" to include that it's required for aws. (The POST throws an error if it's not included when provider is aws.)
